hazelcast/src/main/java/com/hazelcast/query/impl/TypeConverters.java

```diff
@@ -173,8 +184,23 @@ Comparable convertInternal(Comparable value) {
             if (value instanceof BigInteger) {
                 return new BigDecimal((BigInteger) value);
             }
+            if (isNotFloatingPointNumber(value)) {
+                Number number = (Number) value;
+                return BigDecimal.valueOf(number.longValue());
+            }
+            if (value instanceof Double) {
```

Comment:
Maybe we could also add `value instance of Float` and then do `new BigDecimal(float.doubleValue)`